packages/wrangler/src/api/startDevWorker/ConfigController.ts

```diff
@@ -53,14 +53,16 @@ async function resolveDevConfig(
 	config: Config,
 	input: StartDevWorkerInput
 ): Promise<StartDevWorkerOptions["dev"]> {
-	const auth =
-		input.dev?.auth ??
-		(async () => {
-			return {
-				accountId: await getAccountId(),
-				apiToken: requireApiToken(),
-			};
-		});
+	const auth = async () => {
+		if (input.dev?.auth) {
+			return unwrapHook(input.dev.auth, config);
+		}
+
+		return {
+			accountId: config.account_id ?? (await getAccountId()),
```

Comment:
fix default for `startWorker`

Reply:
Does this mean config has a precedent over the env variable?

Reply:
This should probably use the `requireAuth()` helper

Reply:
> This should probably use the `requireAuth()` helper

https://github.com/cloudflare/workers-sdk/pull/6963/commits/5396ab50544827050e2bdb51408368da207c80a5

> Does this mean config has a precedent over the env variable?

Yes. Which matches previous behaviour unless I'm mistaken?

Reply:
> Yes. Which matches previous behaviour unless I'm mistaken

So it does, how surprising

Reply:
I don't think it's surprising. IMO order of precedence: cli args > config file > env vars, because env vars can be set across projects, config file per project, cli args per command – the specificity is clear.

I can understand setting an env var inline and expecting that to take precedence (eg `CLOUDFLARE_ACCOUNT_ID=1234 npx wrangler deploy`) but instead the cli arg should be used for that use-case (eg. `npx wrangler deploy --account-id 1234`) because that env var could've come from the eg ~/.zshrc file

Reply:
I've never seen `--account-id` before - it doesn't seem to be documented 🙃

Reply:
Huh you're right! We should add one to support per command overrides with higher precedence than the config file